src/Select.elm

```diff
@@ -3760,17 +4317,53 @@ resetState (State state_) =
         { state_
             | menuOpen = False
             , activeTargetIndex = 0
-            , menuViewportFocusNodes = Nothing
             , menuListScrollTop = 0
             , menuNavigation = Mouse
             , headlessEvent = Nothing
         }
 
 
-enterSelectTargetItem : SelectState -> List (MenuItem item) -> Maybe (MenuItem item)
+getTargetItem : Int -> MenuItem item -> ( Int, Maybe (MenuItem item) ) -> ListExtra.Step ( Int, Maybe (MenuItem item) )
+getTargetItem targetIndex item ( idx, _ ) =
+    case item of
+        Basic mi ->
+            case mi.virtualConfig of
+                Just cfg ->
+                    if targetIndex == cfg.index then
+                        ListExtra.Stop ( idx, Just item )
+
+                    else
+                        ListExtra.Continue ( idx + 1, Nothing )
+
+                _ ->
+                    if targetIndex == idx then
+                        ListExtra.Stop ( idx, Just item )
+
+                    else
+                        ListExtra.Continue ( idx + 1, Nothing )
+
+        Custom mi ->
+            case mi.virtualConfig of
+                Just cfg ->
+                    if targetIndex == cfg.index then
+                        ListExtra.Stop ( idx, Just item )
+
+                    else
+                        ListExtra.Continue ( idx + 1, Nothing )
+
+                _ ->
+                    if targetIndex == idx then
+                        ListExtra.Stop ( idx, Just item )
+
+                    else
+                        ListExtra.Continue ( idx + 1, Nothing )
+
+
+enterSelectTargetItem : SelectState -> MenuItems item -> Maybe (MenuItem item)
 enterSelectTargetItem state_ viewableMenuItems =
     if state_.menuOpen && not (List.isEmpty viewableMenuItems) then
-        ListExtra.getAt state_.activeTargetIndex viewableMenuItems
+        ListExtra.stoppableFoldl (getTargetItem state_.activeTargetIndex) ( 0, Nothing ) viewableMenuItems
```

Comment:
`stoppableFoldl` is freaking awesome!! Thank you kind person who implemented this!